src/main/SubtitleSelect.tsx

```diff
@@ -28,36 +28,26 @@ import { createTheme, ThemeProvider } from '@mui/material/styles';
 
   // Update the displayFlavors and canBeAddedFlavors
   useEffect(() => {
+    let languages = { ...settings.subtitles.languages };
+
+    // Get flavors of already created tracks or existing subtitle tracks
+    let subtitleFlavors = captionTracks
+      .map(track => track.flavor.type + '/' + track.flavor.subtype)
+      .filter(flavor => !subtitles[flavor])
+      .concat(Object.keys(subtitles));
     let tempDisplayFlavors = []
-    let tempCanBeAddedFlavors = []
-
-    for (let lan in settings.subtitles.languages) {
-      let found = false
-      let subFlavor = lan // left side
-      let name = settings.subtitles.languages[lan] // right side
-
-      // Check if flavor already exists in the tracks from Opencast
-      for (const cap of captionTracks) {
-        if (cap.flavor.type+"/"+cap.flavor.subtype === subFlavor) {
-          found = true
-        }
-      }
-
-      // Need to check this in case of added/deleted subtitles
-      // (aka changes) that are not yet published to Opencast
-      for (const identifier in subtitles) {
-        if (identifier === subFlavor) {
-          found = true
-        }
-      }
-
-      if (found) {
-        tempDisplayFlavors.push({subFlavor: subFlavor, title: name})
-      } else {
-        tempCanBeAddedFlavors.push({subFlavor: subFlavor, title: name})
-      }
+    for (const flavor of subtitleFlavors) {
+      const lang = flavor.replace(/^.*\+/, '') || 'Generic';
+      tempDisplayFlavors.push({
+        subFlavor: flavor,
+        title: languages[flavor] || lang});
+      delete languages[flavor];
     }
 
+    // List of unused languages
+    let tempCanBeAddedFlavors = Object.keys(languages)
+      .map(flavor => ({subFlavor: flavor, title: languages[flavor]}));
+
     setDisplayFlavors(tempDisplayFlavors)
```

Comment:
Maybe we want to sort the flavors before setting them? Before, they where implicitly ordered by their order in the config file, due to iterating over `settings.subtitles.languages`. Now that implicit ordering is gone, it seems the ordering is less consistent. For example when viewing the subtitle select UI, if I have a subtitle track called "English" and add "Deutsch", "Deutsch" will be added to the right of "English". If I click on "Deutsch" and go back to the select view, it will now be to the left of "English".

Reply:
Sort by `title`?

Reply:
Yes, that seems reasonable.